From 1fd7a15f212937b3fd5d02d05536aa66cc78e455 Mon Sep 17 00:00:00 2001
From: Pierre-Hugues Husson <phh@phh.me>
Date: Wed, 26 Oct 2022 17:59:11 -0400
Subject: [PATCH] Fix support for devices without cgroupv2 support

This is technically a revert of 1bd1746447.
The warning inside the commit doesn't really apply to us, because the
devices without cgroupv2 simply won't freeze processes at all
(which is in itself an issue, but a different one)
---
 init/service.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/system/core/init/service.cpp b/system/core/init/service.cpp
index dbae8a771453..2d2b12e80b05 100644
--- a/system/core/init/service.cpp
+++ b/system/core/init/service.cpp
@@ -739,12 +739,15 @@ Result<void> Service::Start() {
                          limit_percent_ != -1 || !limit_property_.empty();
         errno = -createProcessGroup(uid(), pid_, use_memcg);
         if (errno != 0) {
+            PLOG(ERROR) << "createProcessGroup(" << uid() << ", " << pid_ <<  ") failed for service '" << name_ << "'";
+#if 0
             Result<void> result = cgroups_activated.Write(kActivatingCgroupsFailed);
             if (!result.ok()) {
                 return Error() << "Sending notification failed: " << result.error();
             }
             return Error() << "createProcessGroup(" << uid() << ", " << pid_ << ", " << use_memcg
                            << ") failed for service '" << name_ << "': " << strerror(errno);
+#endif
         }
 
         // When the blkio controller is mounted in the v1 hierarchy, NormalIoPriority is
